src/service/resource.js

```diff
@@ -471,10 +472,11 @@ export class Resource {
     }
     this.isFixed_ = isFixed;
 
-    if (isFixed) {
+    if (isFixed && getMode(this.hostWin).runtime != 'inabox') {
```

Comment:
can we change the `isFixed` calculation instead? as for case of inabox, no element can be ever be fixed on screen.

Reply:
Some background on the issue for Justin:
- inabox loads its own viewport service based on the viewport information of the outer window
- resource uses the same viewport service and so when it attempts to subtract off the viewport scroll position, it is using the wrong window's scroll offset causing the issue

I see it we can handle this in a couple of ways:
1.  Assume that nested windows won't scroll and simply ignore subtracting the scroll as I've done here
2.  When inabox, get the window scroll offset from the current window instead of viewport service.  This feels brittle
3.  We could not determine if the element is fixed as Hongfei suggests but other portions of the code may care about this.isFixed_ given its made available via public getter and what I have has the same net effect.

Thoughts?

Reply:
(3) semantically fixes the isFixed logic.
Also I think it's a safe change right now, given the public method `isFixed()` is only used in one place, `resources-impl.js::getResourcesInRect`, which is then used for `iniLoad` load signal.

Reply:
Sure I'm fine with making that change.  @jridgewell can you confirm that meets your needs?  Thanks

Reply:
I still don't understand. Is the element fixed position or not? If so, changing in the definition of here will not work in Layers, which completely changes calculations when hitting a fixed position element.

It sounds like you're pretending this element's rectangle moves as the page scrolls? Why not just define the viewport window without an initial negative offset. As the page scrolls, the `scrollTop - offset` gets closer to zero until the element is in the page.